vm/systemSmartContracts/validator.go

```diff
@@ -1124,9 +1127,35 @@ func (v *validatorSC) unStakeNodesFromStakingSC(blsKeys [][]byte, registrationDa
 		}
 
 		numSuccess++
+
+		stakedData, err := v.getStakedData(blsKey)
+		if err != nil {
+			continue
+		}
+		if stakedData.UnStakedNonce > currentNonce {
+			numSuccessFromWaiting++
+		}
+	}
+
+	numSuccessFromActive := numSuccess - numSuccessFromWaiting
+	return numSuccessFromActive, numSuccessFromWaiting
+}
+
+func (v *validatorSC) computeProcessUnStakeFromNodes(
+	registrationData *ValidatorDataV2,
+	validatorConfig ValidatorConfig,
+	numNodes uint64,
+	unStakeNonce uint64,
+) vmcommon.ReturnCode {
+	if numNodes == 0 {
+		return vmcommon.Ok
+	}
+	unStakeFromActiveNodes := big.NewInt(0).Mul(validatorConfig.NodePrice, big.NewInt(0).SetUint64(numNodes))
```

Comment:
unStakeFromNodes as you call this method either with numSuccessFromWaiting or with numSuccessFromActive

Reply:
done